portfolio/src/main/java/com/google/sps/servlets/DeleteDataServlet.java

```diff
@@ -0,0 +1,44 @@
+package com.google.sps.servlets;
+
+import java.io.IOException;
+
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.DatastoreService;
+
+@WebServlet("/delete")
+public class DeleteDataServlet extends HttpServlet{
+  
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private String entityKind = "Comment";
+  private Key homepageCommentKey = KeyFactory.createKey("Comments", "homepage comments");
+  private String deleteAllParam = "-1";
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+    if (request.getParameter("key").equals(deleteAllParam)) {
+      // Delete all comments data
+      Query query = new Query(entityKind).setAncestor(homepageCommentKey).setKeysOnly();
+      PreparedQuery resultKeys = datastore.prepare(query);
+
+      for (Entity entity : resultKeys.asIterable()) {
+        datastore.delete(entity.getKey());
+      }
+    }
+
+    else {
+      Key key = KeyFactory.stringToKey(request.getParameter("key"));
```

Comment:
i'm not sure myself, but how will stringToKey behave if the key does not exist?

Reply:
it looks like it throws an illegal argument exception (https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/KeyFactory.html#stringToKey-java.lang.String-) -- how would you suggest going about handling this exception?

Reply:
one way is to ignore the request